### ISSUE TITLE
NPM start issue

### ISSUE THREAD

**Issue by akashdave11** [lissy93/twitter-sentiment-visualisation#9]
After starting the mongodb command and running the **npm start** this is coming up

> twitter-sentiment-visualisation@0.0.1 start C:\New folder (2)\Wiki article\twitter-sentiment-visualisation
> node ./bin/www

module.js:471
    throw err;
    ^

Error: Cannot find module 'twit'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\utils\streamer.js:4:10)
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\utils\streamer.js:91:4)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\app.js:52:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! twitter-sentiment-visualisation@0.0.1 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the twitter-sentiment-visualisation@0.0.1 start script 'node ./bin/www'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the twitter-sentiment-visualisation package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./bin/www
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs twitter-sentiment-visualisation
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls twitter-sentiment-visualisation
npm ERR! There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\atul dave\AppData\Roaming\npm-cache\_logs\2017-04-20T17_33_28_916Z-debug.log

Not sure what 'Twit' module is since it is not in the package.json folder

**Comment by lissy93** [comment]
Run `npm install twit`.

**Comment by akashdave11** [comment]
@Lissy93 

Ran the same but the following error came up after "npm start" regarding watsoncredentials username 

`TypeError: Cannot read property 'username' of undefined
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\routes\tone-api.js:16:32)
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\routes\tone-api.js:72:4)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\app.js:74:25)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\bin\www:7:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! twitter-sentiment-visualisation@0.0.1 start: `node ./bin/www`
npm ERR! Exit status 1`

**Comment by lissy93** [comment]
You need to paste your Watson API key into `keys.json`

**Comment by akashdave11** [comment]
@Lissy93 

In the keys.json file i have inserted my twitter api tokens since its asking for the 4 tokens. No mention of the Watson credentials.

Also in the wiki it says to get the api keys by following the links below but the links aren't available

**Comment by lissy93** [comment]
Sorry never added in the links, but you  can just Google it to find URL's.

**HP Haven OnDemand**: http://www.havenondemand.com/login.html?signUp=true
**IBM Watson**: https://www.ibm.com/account/us-en/signup/register.html?Target=https://myibm.ibm.com/
**Google Developer**: https://console.developers.google.com/apis/dashboard
**Twitter Developer**: https://dev.twitter.com/resources/signup

**Comment by akashdave11** [comment]
@Lissy93 

Thank you so much for the links.

So i have obtained the Api keys from the following services

1. Twitter dev 
2. Google Places web api
3. Hp haven ondemand
4. Watson Tone analyzer

However i am confused as to where i have to use these set of keys since the config/keys.coffee only asks for the Twitter Keys i believe. It would really be helpful if you could guide me on that front.

Sorry for all the trouble i have been giving you lately. I am new to these concepts and i am trying to learn about them more. If it runs successfully please allow me to update the documents with the changes that are needed.

**Comment by lissy93** [comment]
No worries, it's no trouble. 

So your `config/keys.json` should look like [this](https://github.com/Lissy93/twitter-sentiment-visualisation/blob/dev/server-side-source/config/sample-keys.coffee) (though with your keys instead of `xxxxxx`, obviously 😜 )
```
exports.twitter =
  consumer_key : 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
  consumer_secret : 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
  token: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
  token_secret: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'

exports.googlePlaces = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'

exports.hp = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'

exports.watson =
  username: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
  password: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
```

I need to sort out the install/setup process to make it quicker for people, it's on my todo list.

**Comment by akashdave11** [comment]
@Lissy93 

Do i need to update both keys.json file? because there are 2 one is config/keys.coffee and the other server-side-source/config/keys.coffee. Since you updated the latter i have only updated server-side-source one. Do i need to update the former one as well with the same format because the format differs in both of them.

After updating only the server-side one it shows the following error
Regarding username and password for watson : https://pastebin.com/D4aBSRUn
 
I had to get the credentials for tone analyzer right?

Also some error come up when performing "npm build run" not sure if they are of any concern but here is the link https://pastebin.com/JjsV3kv7

**Comment by lissy93** [comment]
The source is `server-side-source/config/keys.json`, and the destination is `config/keys.json`
So basically the app reads from `config/keys.json`, but this should have been populated, during the build process with whatever you put in `server-side-source/config/keys.json`. Make sense?

**Comment by akashdave11** [comment]
@Lissy93 

Yup just checked it updated itself automatically. But i am still getting the  error below

C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\node_modules\watson-developer-cloud\lib\index.js:91
            throw new Error('Argument error: api_key or username and password were not specified');
            ^

Error: Argument error: api_key or username and password were not specified
    at Object.tone_analyzer (C:\New folder (2)\Wiki article\twitter-sentiment-visualisation\node_modules\watson-developer-cloud\lib\index.js:91:19)

**Comment by lissy93** [comment]
ok, several things to try:
- In `server-side-source\api-routes\tone-api.coffee` near line 10,  console.log `watsonCredentials` and check it is actually what you expect.
- Check the Watson documentation, that nothing has changed since it came out of beta. Here's the node module that was used: https://www.npmjs.com/package/watson-developer-cloud
- Double check your credentials are actually correct, and there isn't a trailing space or anything in one of the keys.

My guess is that whatever is going wrong will be found out though doing one of those three things

**Comment by lissy93** [comment]
Also with regard to the other link you pasted, with lots of lint warnings. Odd because I don't remember seeing them on my system, but they are only warnings- so shouldn't break the build, or affect the running app. They are just helpful for implementing neater code.

**Comment by akashdave11** [comment]
@Lissy93 

On examining the file i notice that the url is : https://gateway.watsonplatform.net/tone-analyzer-beta/api/

However on my bluemix credential it is : https://gateway.watsonplatform.net/tone-analyzer/api
not the beta one.

Should i change it to non-beta one in `tone-api.coffee`?

**Comment by lissy93** [comment]
Ah, that was a fix that @ianmyles  implemented [in this pull request](https://github.com/Lissy93/twitter-sentiment-visualisation/pull/10), sorry I thought you had pulled since that had been merged. 
The URL should no-longer be the Beta version, yes change it, or re-pull from dev.

**Comment by akashdave11** [comment]
@Lissy93 

I updated the file accordingly and the problem was the indentation of my password key. Thank you for the change.

I ran the mongodb server along with npm start and it shows me the homepage. When i searched for a query the following error emerged

`events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:370:17)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! twitter-sentiment-visualisation@0.0.1 start: `node ./bin/www`
npm ERR! Exit status 1`


update : tried it a second time and it worked except for certain components which were stuck on loading screen(eg entity extraction)

On opening other tabs such as word cloud or entity extraction it stops running. I'll look into it further just wanted to ask whether they were normal
